spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -429,57 +315,55 @@ public void setZkConnectionTimeout(int zkConnectionTimeout) {
 		// The reference point, if not set explicitly is the latest time for anonymous subscriptions and the
 		// earliest time for group subscriptions. This allows the latter to receive messages published before the group
 		// has been created.
-		long referencePoint = this.startOffset != null ?
-				startOffset.getReferencePoint() : (anonymous ? OffsetRequest.LatestTime() : OffsetRequest.EarliestTime());
+		long referencePoint = properties.getStartOffset() != null ?
+				properties.getStartOffset().getReferencePoint() : (anonymous ? OffsetRequest.LatestTime() : OffsetRequest.EarliestTime());
 		return createKafkaConsumer(name, inputChannel, properties, consumerGroup, referencePoint);
 	}
 
+
 	@Override
-	public Binding<MessageChannel> bindProducer(final String name, MessageChannel moduleOutputChannel, Properties properties) {
+	public Binding<MessageChannel> doBindProducer(final String name, MessageChannel moduleOutputChannel, KafkaProducerProperties properties) {
```

Comment:
you mean the class or the parameter? we could - I'm not a huge fan of the unnecessary verbosity. The first parameter is final because we use it in in inner class, but the other ones aren't used that way.